### PR TITLE
MAKE-747: legacy merged mgo queue groups

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: queue/driver_group_mgo.go [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+	session.SetSafe(&mgo.Safe{WMode: "majority"})
+
+	d.mu.Lock()
+	d.session = session
+	d.mu.Unlock()
+
+	startAt := time.Now()
+	go func() {
+		<-dCtx.Done()
+		grip.Info(message.Fields{
```
Comment (bsamek) [inline queue/driver_group_mgo.go]: Is this debugging, or does it really matter?

File: queue/driver_group_mongo.go [hunk 2]
```diff
@@ -184,6 +184,8 @@ func (d *mongoGroupDriver) Get(ctx context.Context, name string) (amboy.Job, err
 		return nil, errors.Wrapf(err, "GET problem fetching '%s'", name)
 	}
 
+	j.Name = j.Name[len(d.group)+1:]
```
Comment (bsamek) [inline queue/driver_group_mongo.go]: I don't think you want `+1` here https://play.golang.org/p/Mkm5nRuBIms

File: queue/driver_group_mongo.go [hunk 3]
```diff
@@ -442,6 +443,7 @@ func (d *mongoGroupDriver) Next(ctx context.Context) amboy.Job {
 			// try for the next thing in the iterator if we can
 			continue
 		}
+		j.Name = j.Name[len(d.group)+1:]
```
Comment (bsamek) [inline queue/driver_group_mongo.go]: `+1`

File: queue/group_remote_mgo_single.go [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+		queues:   map[string]amboy.Queue{},
+	}
+
+	if opts.PruneFrequency > 0 {
+		go func() {
+			pruneCtx, pruneCancel := context.WithCancel(context.Background())
+			defer pruneCancel()
+			ticker := time.NewTicker(opts.PruneFrequency)
+			defer ticker.Stop()
+			defer recovery.LogStackTraceAndContinue("panic in remote queue group ticker")
```
Comment (bsamek) [inline queue/group_remote_mgo_single.go]: I think this defer should be at the top of the goroutine.

File: queue/group_remote_mgo_single.go [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+				}
+			}
+		}()
+	}
+
+	if opts.BackgroundCreateFrequency > 0 {
+		go func() {
+			ticker := time.NewTicker(opts.BackgroundCreateFrequency)
+			defer ticker.Stop()
+			defer recovery.LogStackTraceAndContinue("panic in remote queue group ticker")
```
Comment (bsamek) [inline queue/group_remote_mgo_single.go]: recovery at top of goroutine

File: queue/driver_group_mongo.go (hunk 2 above)
Comment (tychoish) [reply queue/driver_group_mongo.go]: they're dot seperated, so I want to get the dot too.

File: queue/driver_group_mgo.go (hunk 1 above)
Comment (tychoish) [reply queue/driver_group_mgo.go]: debugging

**Pull request merged**